src/types.ts

```diff
@@ -1,3 +1,36 @@
+import { render } from "@testing-library/react";
+import { mount } from "enzyme";
+
+export type FullProps<C extends React.ComponentType> = React.ComponentProps<C>;
+
+type RequiredKeys<T> = {
+  [K in keyof T]-?: Record<string, unknown> extends { [P in K]: T[K] } ? never : K;
+}[keyof T];
+
+export type HasRequiredField<T> = RequiredKeys<T> extends never ? false : true;
+
+export type RenderEnzyme<C extends React.ComponentType, P extends Partial<FullProps<C>>> = (
```

Comment:
For places where names are meaningful (e.g. `C` and `P` here), would you be up for using more full names in generics? I've found it to be difficult to read the single letter abbreviations both here and in general.

Reply:
You got it.